flaskoidc/config.py

```diff
@@ -1,51 +1,43 @@
 import os
 
-OIDC_ATTR_KEY = 'OIDC_'
-SQLALCHEMY_ATTR_KEY = 'SQLALCHEMY_'
-FLASK_SESSION_ATTR_KEY = 'SESSION_'
-EXCEPTIONAL_KEYS = ["OVERWRITE_REDIRECT_URI"]
-LIST_KEYS = ["OIDC_SCOPES"]
 
+class OIDC_PROVIDERS:
+    GOOGLE = 'google'
+    OKTA = 'okta'
+    KEYCLOAK = 'keycloak'
 
-class OIDCProvider:
-    GOOGLE = "GOOGLE"
-    KEYCLOAK = "KEYCLOAK"
+
+# All the custom configurations required for an OIDC provider to work with Authlib
+# will go in here. We'll pass these while registering a client.
+_CONFIGS = {
+    OIDC_PROVIDERS.GOOGLE: {
+        'authorize_params': {
+            'access_type': 'offline',
+            'prompt': 'consent'
+        }
+    },
+    OIDC_PROVIDERS.OKTA: {
+
+    },
+    OIDC_PROVIDERS.KEYCLOAK: {
+
+    }
+}
 
 
 class BaseConfig(object):
-    SECRET_KEY = os.environ.get('FLASK_OIDC_SECRET_KEY', 'base-flask-oidc-secret-key')
+    SECRET_KEY = os.environ.get('FLASK_OIDC_SECRET_KEY', '!-flask-oidc-secret-key')
     WHITELISTED_ENDPOINTS = os.environ.get('FLASK_OIDC_WHITELISTED_ENDPOINTS',
                                            "status,healthcheck,health")
 
-    # Not being used anywhere
-    OIDC_PROVIDER = os.environ.get('FLASK_OIDC_OIDC_PROVIDER', OIDCProvider.KEYCLOAK)
+    OIDC_PROVIDER = os.environ.get('FLASK_OIDC_PROVIDER_NAME', 'google')
+    OIDC_SCOPES = os.environ.get('FLASK_OIDC_SCOPES', 'openid email profile')
+    USER_ID_FIELD = os.environ.get('FLASK_OIDC_USER_ID_FIELD', 'email')
+    CLIENT_ID = os.environ.get('FLASK_OIDC_CLIENT_ID', '')
+    CLIENT_SECRET = os.environ.get('FLASK_OIDC_CLIENT_SECRET', '')
+    REDIRECT_URI = os.environ.get('FLASK_OIDC_REDIRECT_URI', '/auth')
+    CONFIG_URL = os.environ.get('FLASK_OIDC_CONFIG_URL', '')
 
-    # Logging Settings
-    LOG_FORMAT = '%(asctime)s.%(msecs)03d [%(levelname)s] %(module)s.%(funcName)s:%(lineno)d (%(process)d:' \
-                 + '%(threadName)s) - %(message)s'
-    LOG_DATE_FORMAT = os.environ.get('FLASK_OIDC_LOG_DATE_FORMAT', '%Y-%m-%dT%H:%M:%S%z')
-    LOG_LEVEL = os.environ.get('FLASK_OIDC_LOG_LEVEL', 'INFO')
-
-    # OIDC Settings
-    OIDC_CLIENT_SECRETS = os.environ.get('OIDC_CLIENT_SECRETS', 'config/client_secrets.json')
-    OIDC_INTROSPECTION_AUTH_METHOD = os.environ.get('INTROSPECTION_AUTH_METHOD',
-                                                    'client_secret_post')
-    OIDC_ID_TOKEN_COOKIE_SECURE = os.environ.get('OIDC_ID_TOKEN_COOKIE_SECURE', False)
-
-    # Database and Sessions Settings
-    SESSION_TYPE = os.environ.get('SESSION_TYPE', 'sqlalchemy')
     SQLALCHEMY_TRACK_MODIFICATIONS = os.environ.get('SQLALCHEMY_TRACK_MODIFICATIONS', False)
-
     SQLALCHEMY_DATABASE_URI = os.environ.get("SQLALCHEMY_DATABASE_URI", 'sqlite:///sessions.db')
```

Comment:
I am not 100% sure but earlier implementation caused the issue in token when there was redeployment because of sqlite db. But now with Authlib, this issue doesn't come. what exactly is the use case of storing tokens, as browser as well stores the token and uses it  if it's available (Assuming)